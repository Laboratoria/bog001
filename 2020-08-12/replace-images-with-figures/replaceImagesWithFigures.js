const replaceImagesWithFigures = (container) => {
  // const imgs = [...container.querySelectorAll('img')];
  const imgs = container.querySelectorAll('img');

  for (let img of imgs) {
    if (img.parentElement.tagName ==='FIGURE') {
      continue;
    }
    const figure = document.createElement('figure');
    const figcaption = document.createElement('figcaption');
    figcaption.textContent = img.alt;
    img.replaceWith(figure);
    figure.appendChild(img);
    figure.appendChild(figcaption);
  }
};

export default replaceImagesWithFigures;