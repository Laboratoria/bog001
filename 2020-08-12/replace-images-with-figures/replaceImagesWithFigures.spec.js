import replaceImagesWithFigures from './replaceImagesWithFigures.js';

describe('replaceImagesWithFigures', () => {
  it('debería reemplazar imagen', () => {
    const el = document.createElement('div');
    const img = document.createElement('img');
    img.src = 'omg.png';
    img.alt = 'Blah';
    el.appendChild(img);

    replaceImagesWithFigures(el);

    expect(el.children.length).toBe(1);
    expect(el.children[0].tagName).toBe('FIGURE');
    expect(el.children[0].children.length).toBe(2);
    expect(el.children[0].children[0].tagName).toBe('IMG');
    expect(el.children[0].children[1].tagName).toBe('FIGCAPTION');
    expect(el.children[0].children[1].textContent).toBe('Blah');
  });

  it('debería ignorar imagen que ya está dentro de un figure', () => {
    const el = document.createElement('div');
    el.innerHTML = `
      <div>
        <div>
          <figure>
            <img src="omg.png" alt="Blah blah blah" />
          </figure>
        </div>
      </div>
    `;

    expect(el).toMatchSnapshot();
    replaceImagesWithFigures(el);
    expect(el).toMatchSnapshot();
  });
});