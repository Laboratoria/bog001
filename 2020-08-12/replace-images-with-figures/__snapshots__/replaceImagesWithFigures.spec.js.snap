// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaceImagesWithFigures debería ignorar imagen que ya está dentro de un figure 1`] = `
<div>
  
      
  <div>
    
        
    <div>
      
          
      <figure>
        
            
        <img
          alt="Blah blah blah"
          src="omg.png"
        />
        
          
      </figure>
      
        
    </div>
    
      
  </div>
  
    
</div>
`;

exports[`replaceImagesWithFigures debería ignorar imagen que ya está dentro de un figure 2`] = `
<div>
  
      
  <div>
    
        
    <div>
      
          
      <figure>
        
            
        <img
          alt="Blah blah blah"
          src="omg.png"
        />
        
          
      </figure>
      
        
    </div>
    
      
  </div>
  
    
</div>
`;
